### PR TITLE
feat(consensus): [CON-1396] Purge equivocation proofs below and at finalized height

### PR DESCRIPTION
**Pull request by dist1ll**
Equivocation proofs are not needed at `h <= finalized_height`. Purging them more aggressively gives us a better upper bound for the artifact counts.

### CODE REVIEW

**Review by eichhorl — APPROVED** [review]

**Comment by dist1ll** [comment]
Thanks :)

**Comment by dist1ll** [comment]
@eichhorl small change: added a check to only create the change action if there are actually equivocation proofs in the pool. https://github.com/dfinity/ic/pull/927/files#diff-7e30a86343a6b8476c0d1c79c2687b7387ae15638e16aed770b63ed3becbc701R301-R306

File: rs/consensus/src/consensus/purger.rs [hunk 1]
```diff
@@ -287,6 +290,32 @@ impl Purger {
...
+        finalized_height: Height,
+        changeset: &mut ChangeSet,
+    ) {
+        if pool
+            .pool()
+            .validated()
+            .equivocation_proof()
+            .height_range()
+            .is_some()
+        {
```
Comment (eichhorl) [inline rs/consensus/src/consensus/purger.rs:307]: Why is this needed? As I understand, when doing a `PurgeValidatedOfTypeBelow`, we anyway have an early return if the `Meta` table for the type doesn't exist (which is the same we are doing in this check here).

Comment (dist1ll) [reply rs/consensus/src/consensus/purger.rs:307]: It's not needed, but it does add unnecessary change actions into the set, which then show up in unrelated tests. Especially because equivocations are rare, I thought it might be better to only create change actions if they exist. 

But I'm open to getting rid of it as well, and merge the PR as it was before. Just my $0.02.

Comment (eichhorl) [reply rs/consensus/src/consensus/purger.rs:307]: I see, then I think I'm fine with it either way

**Review by eichhorl — APPROVED** [review]

File: rs/consensus/src/consensus/purger.rs (hunk 1 above)
Comment (eichhorl) [reply rs/consensus/src/consensus/purger.rs:307]: I see, then I'm fine with it either way
